nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ListenHTTP.java

```diff
@@ -290,29 +316,23 @@ synchronized private void createHttpServerFromService(final ProcessContext conte
         // get the configured port
         final int port = context.getProperty(PORT).evaluateAttributeExpressions().asInteger();
 
-        final ServerConnector connector;
-        final HttpConfiguration httpConfiguration = new HttpConfiguration();
-        if (keystorePath == null) {
-            // create the connector
-            connector = new ServerConnector(server, new HttpConnectionFactory(httpConfiguration));
-        } else {
-            // configure the ssl connector
-            httpConfiguration.setSecureScheme("https");
-            httpConfiguration.setSecurePort(port);
-            httpConfiguration.addCustomizer(new SecureRequestCustomizer());
+        final ServerConnector connector = createServerConnector(server, port, sslContextService, sslRequired, needClientAuth);
 
-            // build the connector
+        List<Connector> connectors = new ArrayList<>(2);
+        connectors.add(connector);
 
-            connector = new ServerConnector(server, new SslConnectionFactory(contextFactory, "http/1.1"), new HttpConnectionFactory(httpConfiguration));
+        // Add a separate connector for the health check port (if specified)
+        final String healthCheckPortString = context.getProperty(HEALTH_CHECK_PORT).evaluateAttributeExpressions().getValue();
```

Comment:
`asInteger()` could be used. In that way there is no need for the additional string variable.